### PR TITLE
Add maxRetryDuration in the retry logic in spark client; consolidate configs; add unit tests

### PR DESCRIPTION
**Pull request by chakankardb**
In this PR, we are making the following changes:
- Add a maxRetryDurationLimit to the retry logic.
- Retry on socket timeout.
- Refactor/Consolidate spark configs
- Add unit tests.

### CODE REVIEW

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingFileSystem.scala [hunk 1]
```diff
@@ -29,40 +29,19 @@ import org.apache.spark.delta.sharing.{PreSignedUrlCache, PreSignedUrlFetcher}
...
+  lazy private val maxRetryDurationMillis = ConfUtils.maxRetryDurationMillis(getConf)
+  lazy private val timeoutInSeconds = ConfUtils.timeoutInSeconds(getConf)
 
   lazy private val httpClient = {
-    val maxConnections = getConf.getInt("spark.delta.sharing.network.maxConnections", 64)
-    if (maxConnections < 0) {
-      throw new IllegalArgumentException(
-        "spark.delta.sharing.network.maxConnections must not be negative")
-    }
+    val c = getConf()
```
Comment (linzhou-db) [inline spark/src/main/scala/io/delta/sharing/spark/DeltaSharingFileSystem.scala:43]: nit: ``val c` is not used?

File: spark/src/main/scala/io/delta/sharing/spark/util/RetryUtils.scala [hunk 2]
```diff
@@ -27,18 +27,34 @@ private[sharing] object RetryUtils extends Logging {
...
+            "Error during retry attempt " + times + ", retryDuration=" + retryDuration +
+            ", totalDuration=" + totalDuration + " : " + e.getMessage,
+            e
+          )
+          if (shouldRetry(e) && times <= numRetries && totalDuration <= maxDurationMillis) {
+            logWarning(s"Sleeping $sleepMs ms to retry")
+            sleeper(sleepMs)
+            sleepMs *= 2
+          } else {
+            logWarning(s"Not retrying delta sharing rpc")
```
Comment (linzhou-db) [inline spark/src/main/scala/io/delta/sharing/spark/util/RetryUtils.scala:55]: nit: log the exception?

**Review by linzhou-db — APPROVED** [review]

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingFileSystem.scala (hunk 1 above)
Comment (chakankardb) [reply spark/src/main/scala/io/delta/sharing/spark/DeltaSharingFileSystem.scala]: oops, done.

File: spark/src/main/scala/io/delta/sharing/spark/util/RetryUtils.scala (hunk 2 above)
Comment (chakankardb) [reply spark/src/main/scala/io/delta/sharing/spark/util/RetryUtils.scala]: good catch, done.

**Pull request merged**